src/app-simplefin/app-simplefin.js

```diff
@@ -2,150 +2,198 @@ import express from 'express';
 import { inspect } from 'util';
 import https from 'https';
 import { SecretName, secretsService } from '../services/secrets-service.js';
+import { handleError } from '../app-gocardless/util/handle-error.js';
 
 const app = express();
 export { app as handlers };
 app.use(express.json());
 
-app.post('/status', async (req, res) => {
-  let configured = false;
+app.post(
+  '/status',
+  handleError(async (req, res) => {
+    const token = secretsService.get(SecretName.simplefin_token);
+    const configured = token != null && token !== 'Forbidden';
 
-  let token = secretsService.get(SecretName.simplefin_token);
-  if (token != null && token !== 'Forbidden') {
-    configured = true;
-  }
-
-  res.send({
-    status: 'ok',
-    data: {
-      configured: configured,
-    },
-  });
-});
-
-app.post('/accounts', async (req, res) => {
-  let accessKey = secretsService.get(SecretName.simplefin_accessKey);
-
-  if (accessKey == null || accessKey === 'Forbidden') {
-    let token = secretsService.get(SecretName.simplefin_token);
-    if (token == null || token === 'Forbidden') {
+    res.send({
+      status: 'ok',
+      data: {
+        configured: configured,
+      },
+    });
+  }),
+);
+
+app.post(
+  '/accounts',
+  handleError(async (req, res) => {
+    let accessKey = secretsService.get(SecretName.simplefin_accessKey);
+
+    try {
+      if (accessKey == null || accessKey === 'Forbidden') {
+        const token = secretsService.get(SecretName.simplefin_token);
+        if (token == null || token === 'Forbidden') {
+          throw new Error('No token');
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏(comment applicable to all instances of `thow` within this file)

Would you mind either adding a prefix to the message OR creating a new Error class? If these errors happen - they will be printed in the error log. At which point it's easier to follow them if a bit more context is added.

```suggestion
          throw new Error('SimpleFIN: no token provided');
```

Reply:
Even if they are catch?  LIne 44 below catches these errors and actually re-throws a different error (with a better log message).

Reply:
(Or more accurately, it catches these and then returns an error response to the client and does not throw anything.)

Reply:
Good comment. But this raises another point - could we have different error messages for each of the two different errors? The more granularity we have in error handling - the less support tickets we will get with confused users.

Reply:
In general, that is a good point.  In this particular case though, if there is a problem with their access key *or* their setup token the solution is still the same: get a new setup token.  So here I think a single error message makes most sense.